android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java

```diff
@@ -583,25 +589,31 @@ private static boolean putImageInfo(
     }
 
     node.putMap("image", image);
-    return true;
+    return putImageSizeSuccess && putPlayableDurationSuccess;
   }
 
-  private static void putPlayableDuration(
+  /**
+   * @return Whether we succeeded in fetching and putting the playableDuration
+   */
+  private static boolean putPlayableDuration(
       ContentResolver resolver,
       WritableMap image,
       Uri photoUri,
       boolean isVideo,
       boolean includePlayableDuration) {
+    image.putNull("playableDuration");
+
     if (!includePlayableDuration || !isVideo) {
-      image.putNull("playableDuration");
-      return;
+      return true;
     }
 
+    boolean success = true;
```

Comment:
Do you have an example of how/when you would use this returned boolean in user code?

Reply:
I don't think this will ever be called directly user code, because these functions are only called by `putEdges`. `putEdges` is called as part of `getPhotos`.

The previous code for [`putEdges`](https://github.com/react-native-community/react-native-cameraroll/blob/master/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java#L513) skipped over images where `putImageInfo` returned false. This boolean ensures that the same logic applies.

`putImageInfo` used to return false whenever it failed to fetch the `playableDuration` ([line 595](https://github.com/react-native-community/react-native-cameraroll/blob/master/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java#L595)) or the `width`/`height` ([line 612](https://github.com/react-native-community/react-native-cameraroll/blob/master/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java#L612)).

If we fail to fetch either of them, we assume something's wrong with the file and return false. This would cause `putPlayableDurationSuccess` or `putImageSizeSuccess` to be set to false inside of `putImageInfo`. `putImageInfo` would then return false, which would cause the code in `putEdges` to skip the image, just like before.

Reply:
right, that's not a user concern then, thanks for the answer.